validation/gas-token_test.go

```diff
@@ -0,0 +1,105 @@
+package validation
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+func testGasToken(t *testing.T, chain *ChainConfig) {
+	skipIfExcluded(t, chain.ChainID)
+
+	client, err := ethclient.Dial(chain.PublicRPC)
+	require.NoError(t, err, "Failed to connect to the Ethereum client at RPC url %s", chain.PublicRPC)
+	defer client.Close()
+
+	// WETH predeploy name() check
+	want := "0000000000000000000000000000000000000000000000000000000000000020" + // offset
+		"000000000000000000000000000000000000000000000000000000000000000d" + // length
+		"5772617070656420457468657200000000000000000000000000000000000000" // "Wrapped Ether" padded to 32 bytes
+	gotName, err := getString("name()", MustHexToAddress("0x4200000000000000000000000000000000000006"), client)
+	require.NoError(t, err)
+	require.Equal(t, want, gotName)
+
+	// L1Block .isCustomGasToken() check
+	got, err := getBool("isCustomGasToken()", MustHexToAddress("0x4200000000000000000000000000000000000015"), client)
+	if !strings.Contains(err.Error(), "execution reverted") {
+		// Pre: Custom Gas Token feature. Reverting is acceptable.
+		require.NoError(t, err)
+	} else {
+		// Post: Custom Gas Token fearure. Must be set to false.
+		require.False(t, got)
+	}
```

Comment:
Is this right? You're now asserting `got` to be false only if `execution reverted` is in the error, in all other cases you only assert the error to be nil.

Don't you want to make
* if `err != nil` then it must be "exec rev"
* else `got` must be false

same below for the `isCustomGasToken` check

Reply:
[9f02a52](https://github.com/ethereum-optimism/superchain-registry/pull/399/commits/9f02a52aeb2d05b5fb00b8d44179b7d16c2bcc81)